gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/InstallationServiceTest.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Test
+    public void shouldGetInstallation() throws TechnicalException {
+        final InstallationEntity installationEntity = installationService.get();
+
+        verify(installationRepository).find();
+
+        assertNotNull(installationEntity);
+        assertEquals(INSTALLATION_ID, installationEntity.getId());
+        assertEquals(now, installationEntity.getCreatedAt());
+        assertEquals(now, installationEntity.getUpdatedAt());
+        assertEquals(2, installationEntity.getAdditionalInformation().size());
+        assertEquals(COCKPIT_INSTALLATION_ID, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_ID));
+        assertEquals(COCKPIT_INSTALLATION_STATUS, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_STATUS));
+    }
+
+    @Test
+    public void shouldGetInstallationIfExists() throws TechnicalException {
+        final InstallationEntity installationEntity = installationService.getOrInitialize();
+
+        verify(installationRepository).find();
+        verify(installationRepository, times(0)).create(any());
+
+        assertNotNull(installationEntity);
+        assertEquals(INSTALLATION_ID, installationEntity.getId());
+        assertEquals(now, installationEntity.getCreatedAt());
+        assertEquals(now, installationEntity.getUpdatedAt());
+        assertEquals(2, installationEntity.getAdditionalInformation().size());
+        assertEquals(COCKPIT_INSTALLATION_ID, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_ID));
+        assertEquals(COCKPIT_INSTALLATION_STATUS, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_STATUS));
+    }
+
+    @Test
+    public void shouldCreateInstallationIfNotFound() throws TechnicalException {
+        when(installationRepository.find()).thenReturn(Optional.empty());
+        when(installationRepository.create(any())).thenReturn(installation);
+
+        final InstallationEntity installationEntity = installationService.getOrInitialize();
+
+        verify(installationRepository).find();
+        verify(installationRepository).create(any());
+
+        assertNotNull(installationEntity);
+        assertEquals(INSTALLATION_ID, installationEntity.getId());
+        assertEquals(now, installationEntity.getCreatedAt());
+        assertEquals(now, installationEntity.getUpdatedAt());
+        assertEquals(2, installationEntity.getAdditionalInformation().size());
+        assertEquals(COCKPIT_INSTALLATION_ID, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_ID));
+        assertEquals(COCKPIT_INSTALLATION_STATUS, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_STATUS));
+    }
+
+    @Test
+    public void shouldUpdateAdditionalInformation() throws TechnicalException {
```

Comment:
```
Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.021 sec <<< FAILURE! - in io.gravitee.rest.api.service.InstallationServiceTest
shouldUpdateAdditionalInformation(io.gravitee.rest.api.service.InstallationServiceTest)  Time elapsed: 0.005 sec  <<< FAILURE!
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at io.gravitee.rest.api.service.InstallationServiceTest.shouldUpdateAdditionalInformation(InstallationServiceTest.java:151)
```

Reply:
It was hard to reproduce. I've changed the way I compare two dates.